cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraClusterInfo.java

```diff
@@ -64,11 +65,18 @@ public class CassandraClusterInfo implements ClusterInfo, Closeable
     protected transient GossipInfoResponse gossipInfo;
     protected transient CassandraContext cassandraContext;
     protected transient NodeSettings nodeSettings;
+    protected final transient CompletableFuture<List<NodeSettings>> allNodeSettings;
 
     public CassandraClusterInfo(BulkSparkConf conf)
     {
         this.conf = conf;
         this.cassandraContext = buildCassandraContext();
+        this.allNodeSettings = CompletableFuture.supplyAsync(() -> {
```

Comment:
I think it is wasting one thread that is only used to wait for the blocking code to complete. The blocking code is actually a future underlying. `Sidecar.allNodeSettingsBlocking` could be revised to return the future directly, and rename the method.